src/app.js

```diff
@@ -513,36 +510,58 @@ var run = function () {
     loadVersion($('#versionSelector').val());
   });
 
-  // var soljsonSources is provided by bin/list.js
-  if (window.soljsonSources !== undefined) {
-    // populate selector list with available versions
-    $.each(soljsonSources, function (i, file) {
-      if (file) {
-        var version = file.replace(/soljson-(.*).js/, '$1');
-        $('#versionSelector').append(new Option(version, file));
+  $.getJSON('https://ethereum.github.io/solc-bin/bin/list.json', function (data, status) {
+    // loading failed for some reason, fall back to local compiler
+    if (status !== 'success') {
+      $('#versionSelector').append(new Option('latest local version', 'builtin'));
+
+      loadVersion('builtin');
+      return;
+    }
+
+    function buildVersion (build) {
+      if (build.prerelease) {
+        return build.version + '-' + build.prerelease;
+      } else {
+        return build.version;
       }
+    }
+
+    // Sort builds according to semver
+    var builds = data.builds.sort(function (a, b) {
+      a = buildVersion(a);
+      b = buildVersion(b);
+      return semver.compare(a, b);
+    }).reverse();
```

Comment:
I guess instead of `reverse()` I could use `1 - semver.compare(a, b)`, but this way it is more readable.